### DISCUSSION TITLE
Reusable validators

### DISCUSSION THREAD

**Discussion by dotdothu** [pydantic/pydantic#8502] [Question]
I have been using field_validators according to latest documentations (https://docs.pydantic.dev/2.0/usage/validators/#field-validators).

In my application, I have certain relatively complex validation functions that I want to reference across models. One way I can do that is with the decorator. That works (uncommented lines).

However, I can't get the 'reusable validators' working, as the validation does not seem to take place. 

I am sure that I am missing something.

Can you please take a look at this toy example?

```py
def _not_equal_validator(value, expected=0):
    if value == expected:
        raise ValueError(f"the value is {expected}")
    return value

class Example(BaseModel):
    x: int

    field_validator("x")(_not_equal_validator)
    
    #@field_validator('x')
    #def x_validator(cls,value):
    #    _not_equal_validator(value, expected=0)


Example(x=0)
```

**Comment by Iskander14yo** [comment]
Hi! I suppose you need to do following - _assign you validator_ to attribute of class. (as shown here [#940](https://github.com/pydantic/pydantic/issues/940#issuecomment-569765091) ). In your case, it would look like
```python
from pydantic import BaseModel, field_validator, validator

def _not_equal_validator(value, expected=0):
    if value == expected:
        raise ValueError(f"the value is {expected}")
    return value

class Example(BaseModel):
    x: int

    not_equal = field_validator("x")(_not_equal_validator)

    #@field_validator('x')
    #def x_validator(cls,value):
    #    _not_equal_validator(value, expected=0)


Example(x=0)
```

**Comment by Viicos** [comment]
See https://docs.pydantic.dev/latest/concepts/validators/#using-the-annotated-pattern.
